WordPress/Classes/ViewRelated/Reader/ReaderStreamViewController.swift

```diff
@@ -915,13 +915,23 @@ import WordPressFlux
 
         let lastSynced = topic.lastSynced ?? Date(timeIntervalSince1970: 0)
         let interval = Int( Date().timeIntervalSince(lastSynced))
-        if canSync() && (interval >= refreshInterval || topic.posts.count == 0) {
+
+        if canSync() && (interval >= refreshInterval || topicPostsCount == 0) {
             syncHelper?.syncContentWithUserInteraction(false)
         } else {
             handleConnectionError()
         }
     }
 
+    /// Returns the number of posts for the current topic
+    /// This allows the count to be overriden by subclasses
+    var topicPostsCount: Int {
```

Comment:
No need to change necessarily but could have also been a one liner :
`return readerTopic != nil ? readerTopic!.posts.count : 0`

Reply:
Makes sense, although I'm not a fan of the force unwrapping. I would change it to just:

`return readerTopic?.posts.count ?? 0`

Reply:
lol great!